Skeleton/src/WareHouse.cpp

```
@@ -0,0 +1,125 @@
+                int coolDown_maxDistance, maxOrders, distancePerStep;
+                iss >> name >> role >> coolDown_maxDistance;
+                if (role.find("driver") != std::string::npos)
+                {
+                    iss >> distancePerStep;
+                    if (role.find("limited") != std::string::npos)
+                    {
+                        iss >> maxOrders;
+                        Volunteer* driver = new LimitedDriverVolunteer(volunteerCounter+1, name, coolDown_maxDistance, distancePerStep, maxOrders);
+                        volunteers.push_back(driver);
+                    }
+                    else
+                    {
+                        Volunteer* driver = new DriverVolunteer(volunteerCounter+1, name, coolDown_maxDistance, distancePerStep);
+                        volunteers.push_back(driver);
+                    }
+                    
+                } else {
+                    if (role.find("limited") != std::string::npos)
+                    {
+                        iss >> maxOrders;
+                        Volunteer* collector = new LimitedCollectorVolunteer(volunteerCounter+1, name, coolDown_maxDistance, maxOrders);
+                        volunteerCounter++;
+                        volunteers.push_back(collector);
+                    }
+                    else
+                    {
+                        Volunteer* collector = new CollectorVolunteer(volunteerCounter+1, name, coolDown_maxDistance);
+                        volunteerCounter++;
+                        volunteers.push_back(collector);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void WareHouse::addOrder(Order* order) {
+    pendingOrders.push_back(order);
+}
+void WareHouse::addAction(BaseAction* action) {
+    actionsLog.push_back(action);
+}
+Customer& WareHouse::getCustomer(int customerId) const {
+    if(customerCounter<customerId || customerId<1) {throw std::exception();}
```

We should throw an exception? That was instructed in the assignment?